internal/restic/backend.go

```diff
@@ -17,6 +18,9 @@ type Backend interface {
 	// repository.
 	Location() string
 
+	// Hasher may return a hash function for calculating a content hash for the backend
+	Hasher() hash.Hash
```

Comment:
Nitpick, but instead of adding lots of no-ops, maybe check for

```go
type HashingSaver interface {
	Saver
	Hasher() hash.Hash
}
```

in the packer code?

Reply:
Wouldn't that require all the `NewBytesReader` call in code and especially in the tests.

Reply:
Yeah, I guess it might not actually be more elegant.